### PR TITLE
Use ZSERIO_JAR_PATH option if set

### PR DESCRIPTION
**Pull request by johannes-wolf**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,27 @@ set(SRC "${CMAKE_CURRENT_SOURCE_DIR}")
 set(CMAKE_POSITION_INDEPENDENT_CODE ON) # This does not set -fPIC :(
 set(CMAKE_CXX_FLAGS "-fPIC")
 
+## ZSerio binary
+####
+option(ZSERIO_JAR_PATH "Path to zserio jar.")
+if (ZSERIO_JAR_PATH)
+  set(ZSERIO "java")
```
Comment (MisterGC) [inline CMakeLists.txt:14]: Consider using `find_package(Java REQUIRED COMPONENTS Runtime)` and then `Java_JAVA_EXECUTABLE`

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,27 @@ set(SRC "${CMAKE_CURRENT_SOURCE_DIR}")
...
+  list(APPEND ZSERIO "-jar")
+  list(APPEND ZSERIO "${ZSERIO_JAR_PATH}")
+
+  message("Using zserio jar: ${ZSERIO_JAR_PATH}")
+else()
+  find_program(ZSERIO zserio)
```
Comment (MisterGC) [inline CMakeLists.txt:20]: Consider dropping support for a zserio executable completely as this is not the way zserio deployed/published.

**Review by MisterGC — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (KRE02) [reply CMakeLists.txt]: Using `find_program` should be enough, we need java at compile (configure) time only.

**Review by josephbirkner — APPROVED** [review]
Perfect!

**Pull request merged**